packages/loopring_v3/contracts/impl/libexchange/ExchangeWithdrawals.sol

```diff
@@ -37,299 +22,236 @@ library ExchangeWithdrawals
     using AddressUtil       for address payable;
     using BytesUtil         for bytes;
     using MathUint          for uint;
-    using ExchangeAccounts  for ExchangeData.State;
     using ExchangeBalances  for ExchangeData.State;
     using ExchangeMode      for ExchangeData.State;
     using ExchangeTokens    for ExchangeData.State;
 
-    event WithdrawalRequested(
-        uint    indexed withdrawalIdx,
-        uint24  indexed accountID,
-        uint16  indexed tokenID,
-        uint96          amount
+    event ForcedWithdrawalRequested(
+        address indexed owner,
+        address         token,
+        uint24  indexed accountID
     );
 
     event WithdrawalCompleted(
-        uint24  indexed accountID,
-        uint16  indexed tokenID,
-        address         to,
+        address indexed from,
+        address indexed to,
+        address         token,
         uint96          amount
     );
 
     event WithdrawalFailed(
-        uint24  indexed accountID,
-        uint16  indexed tokenID,
-        address         to,
+        address indexed from,
+        address indexed to,
+        address         token,
         uint96          amount
     );
 
-    function getWithdrawRequest(
+    function forceWithdraw(
```

Comment:
It's just the minimum amount of time the exchange needs to keep the exchange out of withdrawal mode. I don't really see any reason why he can't just withdraw immediately after `MIN_TIME_IN_SHUTDOWN` has exceeded if he wants to quit as soon as possible? Disabling `forceWithraw` also means we can't allow the exchange to keep running after `MIN_TIME_IN_SHUTDOWN`, it would be a hard limit which doesn't seem to be a good idea. 

I also don't see `forceWithraw` being used by anyone, even in shutdown mode. it's just there for the worst possible scenario, so doesn't seem necessary to make things more complex then necessary.

Reply:
I did another analysis, seems my previous one was wrong :(